### DISCUSSION TITLE
peristence of pyaedt session through iterations

### DISCUSSION THREAD

**Discussion by gkorompi** [ansys/pyaedt#4670] [Q&A] [answered]
So i have the following script:

def test(var):
    import pyaedt
    import pandas as pd
    hfss = pyaedt.Hfss(path)
    object_names = [*hfss.modeler.objects_by_name.keys()]
    for i in object_names:
         ......................
    hfss.release_desktop()

here the test are modifications manipulated by an algorithm through the var inputs. In order to save time, my thought process is to make the pyaedt session persistent and to not open and close it in each iteration. Nevertheless, pyaedt and hfss should always be defined in order for the following commands to run. Any ideas on how to do this? Thanks!

**Comment by Samuelopez-ansys** [comment]
Hi @gkorompi ,

please describe a bit better the workflow.

I need to understand what “vars” means and its context

**Comment by gkorompi** [comment]
Hello @Samuelopez-ansys,

vars are the variables manipulating my structure, i.e. patch width. The algorithm changes this values in order to achieve the optimum result. My question is, how to keep the same aedt session open (project + design) through iterations (test is called multiple times). Because in the way that i have written it above, a new session is opened and later released in each iteration. Ty

**Reply by Samuelopez-ansys** [reply, marked as answer]
@gkorompi But why do not simply create the pyaedt object at the beginning? I need more context because I would first create the pyaedt object:

```
import pyaedt
import pandas as pd
hfss = pyaedt.Hfss(path)
```
Then:

```
def test(var):
  
    object_names = [*hfss.modeler.objects_by_name.keys()]
    for i in object_names:
        # do your things
```

and there you can call the method.

```
test(1)

test(2)

....

hfss.release_desktop()
```

**Comment by gkorompi** [comment]
Okay yes you are right, i can pass hfss as a variable in the function:

def test(var,hfss):

i thought this was not possible as it would corrupt the session. 

Thanks a lot!
